extensions/amp-date-countdown/1.0/date-countdown.js

```diff
@@ -47,95 +50,104 @@ const TimeUnit = {
 };
 
 // Default prop values
-const DEFAULT_OFFSET_SECONDS = 0;
 const DEFAULT_LOCALE = 'en';
 const DEFAULT_WHEN_ENDED = 'stop';
 const DEFAULT_BIGGEST_UNIT = 'DAYS';
 
+/**
+ * @param {!JsonObject} data
+ * @return {string}
+ */
+const DEFAULT_RENDER = (data) =>
+  /** @type {string} */ (`${data['days']} ${data['dd']}, ` +
+    `${data['hours']} ${data['hh']}, ` +
+    `${data['minutes']} ${data['mm']}, ` +
+    `${data['seconds']} ${data['ss']}, `);
+
 /**
  * @param {!DateCountdownPropsDef} props
  * @return {PreactDef.Renderable}
  */
 export function DateCountdown({
-  endDate,
-  timeleftMs,
-  timestampMs,
-  timestampSeconds,
-  offsetSeconds = DEFAULT_OFFSET_SECONDS,
+  dateTime,
   whenEnded = DEFAULT_WHEN_ENDED,
   locale = DEFAULT_LOCALE,
   biggestUnit = DEFAULT_BIGGEST_UNIT,
-  render,
-  children,
+  render = DEFAULT_RENDER,
   ...rest
 }) {
   useResourcesNotify();
   const {playable} = useAmpContext();
-  const epoch = useMemo(
-    () =>
-      getEpoch(endDate, timeleftMs, timestampMs, timestampSeconds) +
-      offsetSeconds * MILLISECONDS_IN_SECOND,
-    [endDate, timeleftMs, timestampMs, timestampSeconds, offsetSeconds]
+
+  const epoch = getDate(dateTime);
+
+  // How much time is left in our countdown
+  const setTimeleft = useState(epoch - Date.now())[1];
```

Comment:
`const [, setTimeLeft] = useState(...);` works, I've updated to this.

Regarding the structure you mentioned, I think it probably can work, but because of the description below from: https://reactjs.org/docs/hooks-reference.html#usememo

I was more hesitant to use `useMemo` here instead of `useRef` which I am currently using.

You may rely on useMemo as a performance optimization, not as a semantic guarantee. In the future, React may choose to “forget” some previously memoized values and recalculate them on next render, e.g. to free memory for offscreen components. Write your code so that it still works without useMemo — and then add it to optimize performance.

Reply:
Thanks Dima, I've updated to `useMemo` in the latest revision.  Similar to the pattern shown here.